### ISSUE TITLE
React v19 support

### ISSUE THREAD

**Issue by rinarakaki** [connectrpc/connect-query-es#476]
Would we include react v19rc support in the upcoming v2? https://github.com/connectrpc/connect-query-es/pull/475

**Comment by paul-sachs** [comment]
We haven't tested the library with react 19, though it probably won't have any problems assuming react-query doesn't have any react 19 issues. @rinarakaki have you noticed any specific issues?

**Comment by rinarakaki** [comment]
I have a problem as

```
# npm run build

...

   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Linting and checking validity of types    
   Collecting page data  ...Error: Failed to collect configuration for /_not-found
    at <unknown> (/path/to/my-project/node_modules/next/dist/build/utils.js:1131:23)
    at async Span.traceAsyncFn (/path/to/my-project/node_modules/next/dist/trace/trace.js:153:20) {
  [cause]: TypeError: (0 , i.createContext) is not a function
      at 47394 (/path/to/my-project/.next/server/chunks/44076.js:3:104807)
      at t (/path/to/my-project/.next/server/webpack-runtime.js:1:127)
      at 19611 (/path/to/my-project/.next/server/app/_not-found/page.js:1:3703)
      at Function.t (/path/to/my-project/.next/server/webpack-runtime.js:1:127)
      at async getLayoutOrPageModule (/path/to/my-project/node_modules/next/dist/server/lib/app-dir-module.js:33:15)
      at async collectAppPageSegments (/path/to/my-project/node_modules/next/dist/build/segment-config/app/app-segments.js:50:45)
      at async (/path/to/my-project/node_modules/next/dist/build/utils.js:1129:28)
      at async Span.traceAsyncFn (/path/to/my-project/node_modules/next/dist/trace/trace.js:153:20)
}

> Build error occurred
Error: Failed to collect page data for /_not-found
    at <unknown> (/path/to/my-project/node_modules/next/dist/build/utils.js:1234:15) {
  type: 'Error'
}
```

I should also try the latest rc.3 so could you resolve https://github.com/connectrpc/connect-query-es/issues/481?

**Comment by rinarakaki** [comment]
I haven't had any problems so far. Also better to try the latest rc.3 so could you please resolve https://github.com/connectrpc/connect-query-es/issues/481?

**Comment by timostamm** [comment]
Let's reopen this as "React v19 support". 

I think it's possible to relax the peer dependency constraint to include the v19 rc to make it easier for users to try out. 

I'm not sure TanStack Query works well with the new `use` though, see [this blog](https://tkdodo.eu/blog/react-19-and-suspense-a-drama-in-3-acts) post for context. React 19 is still a release candidate, and it's possible that there will be another one with breaking changes.

**Comment by paul-sachs** [comment]
Fixed in [2.0.1](https://github.com/connectrpc/connect-query-es/releases/tag/v2.0.1)

**Comment by rinarakaki** [comment]
Thanks!
